### PR TITLE
vr: fix packet interface mismatch

### LINKED ISSUE #6615: VR reply packets - interface mismatch
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
VR
~~~

##### CLOUDSTACK VERSION
~~~
4.14.1 and onwards
~~~

##### CONFIGURATION
Advanced networking with at least 2 public ranges
Network with one source NAT IP on one range, enable static NAT IP from the other public range

##### OS / ENVIRONMENT
Tested with Vmware 6.7

##### SUMMARY
The outgoing traffic for VMs goes from one VR interface when the request is initiated within the VM, however if the request comes from the internet to the static NAT IP, then the reply goes from a different VR interface

##### STEPS TO REPRODUCE
~~~
- Add an additional public range on the physical network for a zone
- Create a network and deploy a VM on it
- Acquire an additional public IP from the new range on the network
- Enable static NAT on the new IP
- Verify a new VR interface has been created
- (if necessary enable egress rules)
- From within the VM, ping a server outside the network -> Verify the traffic on the VR goes through one interface (for example with tcpdump)
- From outside the network, ping the VM static NAT IP -> Verify the request on VR arrives to the same interface as the step above but the reply is sent through a different interface
~~~

##### EXPECTED RESULTS
~~~
Same interface is used for the outgoing traffic
~~~

##### ACTUAL RESULTS
~~~
root@r-18-VM:~# tcpdump -i eth4 icmp -n
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eth4, link-type EN10MB (Ethernet), snapshot length 262144 bytes
16:23:45.383531 IP 10.0.3.251 > 10.0.80.37: ICMP echo request, id 9332, seq 72, length 64
16:23:46.383363 IP 10.0.3.251 > 10.0.80.37: ICMP echo request, id 9332, seq 73, length 64
16:23:47.394501 IP 10.0.3.251 > 10.0.80.37: ICMP echo request, id 9332, seq 74, length 64
16:23:48.389304 IP 10.0.3.251 > 10.0.80.37: ICMP echo request, id 9332, seq 75, length 64
16:23:49.394640 IP 10.0.3.251 > 10.0.80.37: ICMP echo request, id 9332, seq 76, length 64

root@r-18-VM:~# tcpdump -i eth2 icmp -n
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on eth2, link-type EN10MB (Ethernet), snapshot length 262144 bytes
16:23:53.407090 IP 10.0.80.37 > 10.0.3.251: ICMP echo reply, id 9332, seq 80, length 64
16:23:54.406632 IP 10.0.80.37 > 10.0.3.251: ICMP echo reply, id 9332, seq 81, length 64
16:23:55.414142 IP 10.0.80.37 > 10.0.3.251: ICMP echo reply, id 9332, seq 82, length 64
16:23:56.411103 IP 10.0.80.37 > 10.0.3.251: ICMP echo reply, id 9332, seq 83, length 64
16:23:57.412352 IP 10.0.80.37 > 10.0.3.251: ICMP echo reply, id 9332, seq 84, length 64
~~~

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Fixes #6615

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by weizhouapache** [comment]
@blueorangutan package

File: systemvm/debian/opt/cloud/bin/cs/CsAddress.py [hunk 1]
```diff
@@ -423,7 +423,7 @@ def fw_router(self):
                             "-d %s/32 -j FIREWALL_%s" % (self.address['public_ip'], self.address['public_ip'])])
             self.fw.append(["mangle", "front",
                             "-A FIREWALL_%s " % self.address['public_ip'] +
-                            "-m state --state RELATED,ESTABLISHED -j ACCEPT"])
+                            "-m state --state RELATED,ESTABLISHED -j RETURN"])
```
Comment (weizhouapache) [inline systemvm/debian/opt/cloud/bin/cs/CsAddress.py:426]: it looks the code in systemvm/debian/opt/cloud/bin/configure.py need to be updated as well (search `FIREWALL_`)

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@shwstppr @weizhouapache I rebased this on 4.17 and added a small change to the configure.py code: https://github.com/apache/cloudstack/pull/6656/commits/7010f6a1341643aed036d60aedec9c1af1c6afa2
tested this and it seems correct. please review.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@DaanHoogland code lgtm, but I am not 100% confident about the changes. 

btw, can you kick off smoke tests ?

**Comment by DaanHoogland** [comment]
> @DaanHoogland code lgtm, but I am not 100% confident about the changes.
> 
> btw, can you kick off smoke tests ?

I have hack-tested this, but I'll update my test env with these new packages and retest to make sure, @weizhouapache 
and yes af course ;)
@blueorangutan test

**Comment by weizhouapache** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
> > @DaanHoogland code lgtm, but I am not 100% confident about the changes.
..
> I have hack-tested this, but I'll update my test env with these new packages and retest to make sure, @weizhouapache and yes 

@weizhouapache , I manually checked the ping on the extra interface in the router (eth3) and it contains both incoming and outgoing data. Than I checked doing `ls` in an ssh connection and again both incoming and outgoing data where on eth3

**Comment by DaanHoogland** [comment]
@nvazquez @weizhouapache can you review please?

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@weizhouapache @nvazquez can you shine your lights over this please?

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Review by nvazquez — APPROVED** [review]
Code LGTM

**Comment by DaanHoogland** [comment]
@weizhouapache @shwstppr are we satisfied with this now?

**Review by weizhouapache — APPROVED** [review]
code lgtm

**Pull request merged**